### DISCUSSION TITLE
How to render an empty node?

### DISCUSSION THREAD

**Discussion by Magicloud** [ivanceras/sauron#85] [Q&A] [answered]
For example,. showing a message when it is set.
```rust
if let Some(msg) = self.message {
  node! {
    <div>{text(msg)}</div>
  }
} else {
  // What to do here?
}
```

**Comment by ivanceras** [comment, marked as answer]
You can either go with:

```rust
view_if(self.message.is_some(), {
  node! {
    <div>{text(self.message.unwrap())}</div>
  }
}
```
or

```rust
if let Some(msg) = self.message {
  node! {
    <div>{text(msg)}</div>
  }
} else {
  <span></span>
}
```

**Reply by Magicloud** [reply]
`view_if` could be tricky. The first parameter being false does not prevent the second parameter from being executed. When `unwrap` outside (before) `node`, the whole thing panic when `message` is `None`.

**Reply by ivanceras** [reply]
@Magicloud there is also [`lazy_view_if`](https://docs.rs/sauron/latest/sauron/prelude/html/fn.lazy_view_if.html). I've added this for the same usage as what you are doing.

**Reply by Magicloud** [reply]
Cool. Thanks.

**Comment by flosse** [comment]
Is there a way to map an `Option<Node<M>>` to `Node<M>`? That would be very nice for different situations.
Also `Vec<Node<M>>` would be nice:

```rust
fn foo(mdl: &Mdl) -> Node<M> {
   vec![ ] // would be empty
}
```

**Reply by ivanceras** [reply]
You can try [`node_list`](https://docs.rs/sauron/latest/sauron/html/fn.node_list.html#)
```rust
fn foo(mdl: &Mdl) -> Node<M> {
   node_list([]) // would be empty
}
```

**Reply by flosse** [reply]
Thanks for your fast reply! :pray:

**Comment by vjackson725** [comment]
```
node! { <> </> }
```
also appears to create up an empty node work. I'm not sure what the non-macro version would be.
